include/dlaf/factorization/qr/api.h

```diff
@@ -11,6 +11,7 @@
 #pragma once
 
 #include <complex>
+#include <vector>
```

Comment:
Nit: I suppose you could remove this? It was only for the vector<TileSender>, right? Also in `include/dlaf/factorization/qr.h`?